orion/primitives/tadgan.py

```diff
@@ -82,235 +127,254 @@ def __getstate__(self):
 
         for network in networks:
             with tempfile.NamedTemporaryFile(suffix='.hdf5', delete=False) as fd:
-                keras.models.save_model(state.pop(network), fd.name, overwrite=True)
-
+                tf.keras.models.save_model(state.pop(network), fd.name, overwrite=True)
                 state[network + '_str'] = fd.read()
 
         return state
 
-    def __setstate__(self, state):
+    def __setstate__(self, state: dict):
         networks = ['critic_x', 'critic_z', 'encoder', 'generator']
         for network in networks:
             with tempfile.NamedTemporaryFile(suffix='.hdf5', delete=False) as fd:
                 fd.write(state.pop(network + '_str'))
                 fd.flush()
 
-                state[network] = keras.models.load_model(fd.name)
+                state[network] = tf.keras.models.load_model(fd.name)
 
         self.__dict__ = state
 
-    def _build_model(self, hyperparameters, layers, input_shape):
-        x = Input(shape=input_shape)
-        model = keras.models.Sequential()
+    @classmethod
```

Comment:
Is there a reason this is a class method? I don't have a preference just wondering about the logic.